### ISSUE TITLE
Allow override of existing conversation ID

### ISSUE THREAD

**Issue by ramonsmits** [Particular/NServiceBus#5088]
In https://github.com/Particular/NServiceBus/issues/4171 the focus is to easily set a conversions ID when sending the first message in a conversation.

It would be very convenient to override the conversation ID on an existing conversation. When you currently try to set a new conversation ID you will get the following exception:

> Cannot set the NServiceBus.ConversationId header to 'XXXXX' as it cannot override the incoming header value ('2f4076a0-d8de-4297-9d18-a830015dd42a').

There are reasons for overriding the conversation ID, for example:

- When you have a never ending/ever lasting/very long living saga that has a timeout scheduled every hour/day/week. All messages being sent from that saga share the same conversation ID. In reality the saga only acts like a trigger and you would want to override the conversation ID when that timeout sends/publishes a message.
- When you have an integration message handler that queries a store and for each returned item sends/publishes a message. Here you also want to set the conversation ID to a new value for each item.

**Comment by ramonsmits** [comment]
@DavidBoike wrote a nice blog post on how to write a workaround to still make this possible: https://www.make-awesome.com/2020/04/overriding-the-nservicebus-conversationid/

It should not be mandatory that customers have to use such workarounds in order to set that header.

Maybe its easy to just lift the method that sets this header to something like an "update" to that the API is clear in its intent that is will overwrite an existing value.

```c#
var sendOptions = new SendOptions();
sendOptions.Headers.Update(Headers.ConversationId, "MyCustomConversationId/" + System.Guid.NewGuid(), Header);
```

**Comment by mikeminutillo** [comment]
@ramonsmits a dedicated API would allow us to leave a breadcrumb header for the tooling to follow. It also allows the customer to start a new conversation without having to specify the id. i.e.

```csharp
var sendOptions = new SendOptions();
sendOptions.StartNewConversation();
// OR
sendOptions.StartNewConversation("MyCustomConversationId/" + System.Guid.NewGuid());
```

Under the covers this could override the `ConversationId` header and set a `PreviousConversationId` header that we could use to link visualizations.

**Comment by ramonsmits** [comment]
Available since version 7.4.0, guidance at:

- https://docs.particular.net/nservicebus/messaging/headers#messaging-interaction-headers-nservicebus-conversationid-starting-a-new-conversation
